File: Backend/Domain/Box.cs
namespace Domain;

public class Box
{
    public int Id { get; set; }
    public string Contents { get; set; }
    public double XWidth { get; set; }
    public double ZHeight { get; set; }
    public double YLength { get; set; }
    public double Weight { get; set; }



    public Box()
    {
        
    }
    
    public Box(int id, string contents, double xWidth, double zHeight, double yLength, double weight)
    {
        Id = id;

        if (!String.IsNullOrWhiteSpace(contents))
            Contents = contents;
        else Contents = "Empty Box";
        
        XWidth = xWidth;
        ZHeight = zHeight;
        YLength = yLength;
        Weight = weight;

      
    }
    
    
    
  
}

File: Frontend/SDP-Compulsory-Frontend/src/app/core/box.ts
export class Box {
  id: number = -1;
  contents: string = "";
  xWidth: number = 0;
  yLength: number = 0;
  zHeight: number = 0;

  volume: number = 0;
  density: number = 0;

  constructor(
    id: number,
    contents: string,
    xWidth: number,
    yLength: number,
    zHeight: number
  )
  {
    this.id = id;
    this.contents = contents;
    this.xWidth = xWidth;
    this.yLength = yLength;
    this.zHeight = zHeight;
  }

}
